File: src/java/EntityFX/src/EntityFX/Core/Generic/MathBase.java
package EntityFX.Core.Generic;

import java.io.FileNotFoundException;

import EntityFX.Core.Writer;

public abstract class MathBase<TResult> extends BenchmarkBase<TResult> {

    public MathBase(Writer writer, boolean printToConsole)  throws FileNotFoundException {
        super(writer, printToConsole);
        Iterrations = 200000000;
        Ratio = 0.5;
    }


    protected static double doMath(int i, double li)
    {
        double rev = 1.0 / (i + 1.0);
        return Math.abs(i) * Math.acos(rev) * Math.asin(rev) * Math.atan(rev) +
            Math.floor(li) + Math.exp(rev) * Math.cos(i) * Math.sin(i) * Math.PI + Math.sqrt(i);
    }

}

File: src/dotnet/EntityFX.NetBenchamarks.Core/Generic/MathBase.cs
using System;
using System.Runtime.CompilerServices;

namespace EntityFX.NetBenchmark.Core.Generic
{
    public abstract class MathBase<TResult> : BenchmarkBase<TResult>
    {
        public MathBase(IWriter writer)
            :base(writer)
        {
            Iterrations = 200000000;
            Ratio = 0.5;
        }
#if NETSTANDARD2_0 || NET45
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
#endif
        protected static double DoMath(int i, double li)
        {
            double rev = 1.0 / (i + 1.0);
            return Math.Abs(i) * Math.Acos(rev) * Math.Asin(rev) * Math.Atan(rev) +
                Math.Floor(li) + Math.Exp(rev) * Math.Cos(i) * Math.Sin(i) * Math.PI + Math.Sqrt(i);
        }
    }
}
